astropy/table/operations.py

```diff
@@ -426,6 +449,13 @@ def join(
         keys_right=keys_right,
     )
 
+    if sort_table is not None:
+        # Sort the table back to the original order and remove the temporary columns.
+        # If sort_table is not None that implies keep_order=True.
```

Comment:
so ... why not use `if keep_order:` directly ?

Reply:
For `keep_order=True` and `join_type="outer"` then `sort_table` is `None` and the code would fail.